Yank/multistate/multistateanalyzer.py

```diff
@@ -517,6 +520,60 @@ def read_energies(self):
 
         return energy_matrix, unsampled_energy_matrix, neighborhoods, sampled_states
 
+    @property
+    def has_log_weights(self):
+        """
+        Return True if the storage has log weights, False otherwise
+        """
+        # TODO: Add general API points for reporter so we don't have to access NetCDF variables directly
```

Comment:
You could just call `read_logZ` and trap the error as a way of asking forgiveness rather than permission. But as a quick "we need this out now" its okay